src/main/java/org/alliancegenome/curation_api/services/helpers/validators/GenomicEntityValidator.java

```diff
@@ -37,4 +54,85 @@ public String validateName(GenomicEntity uiEntity) {
 		return name;
 	}
 	
+	public List<Synonym> validateSynonyms(GenomicEntity uiEntity, GenomicEntity dbEntity) {
+		String field = "synonyms";
+		
+		List<Synonym> validatedSynonyms = new ArrayList<Synonym>();
+		if (CollectionUtils.isNotEmpty(uiEntity.getSynonyms())) {	
+			for (Synonym newSynonym : uiEntity.getSynonyms()) {
+				ObjectResponse<Synonym> synonymResponse = synonymValidator.validateSynonym(newSynonym);
+				if (synonymResponse.getEntity() == null) {
+					Map<String, String> errors = synonymResponse.getErrorMessages();
+					for (String synonymField : errors.keySet()) {
+						addMessageResponse(field, synonymField + " - " + errors.get(synonymField));
+					}
+					return null;
+				}
+				validatedSynonyms.add(synonymResponse.getEntity());
+			}
+		}
+		
+		List<Long> previousIds = new ArrayList<Long>();
+ 		if(CollectionUtils.isNotEmpty(dbEntity.getSynonyms()))
+			previousIds = dbEntity.getSynonyms().stream().map(Synonym::getId).collect(Collectors.toList());
+ 		List<Long> validatedIds = new ArrayList<Long>();
+		if (CollectionUtils.isNotEmpty(validatedSynonyms))
+			validatedIds = validatedSynonyms.stream().map(Synonym::getId).collect(Collectors.toList());
+		for (Synonym validatedSynonym : validatedSynonyms) {
+			if (!previousIds.contains(validatedSynonym.getId()))
+				synonymDAO.persist(validatedSynonym);
+		}
+		
+		List<Long> idsToRemove = ListUtils.subtract(previousIds, validatedIds);
+		for (Long id : idsToRemove) {
+			genomicEntityDAO.deleteAttachedSynonym(id);
```

Comment:
Rather than running the hardcoded query you could do something like this:
```java
Predicate<Synonym> removeCondition = synonym -> idsToRemove.contains(synonym.getId());
dbEntity.getSynonyms().removeIf(removeCondition);
```

Reply:
And you may need a DELETE CASCADE on the GE.synonyms field so that you don't have to delete synonyms directly.

Reply:
@oblodgett - good tip, done